api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirMediaDaoImpl.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.dao.impl;
+
+import static org.hibernate.criterion.Restrictions.eq;
+
+import javax.annotation.Nonnull;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import ca.uhn.fhir.rest.param.DateParam;
+import ca.uhn.fhir.rest.param.ReferenceAndListParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
+import lombok.AccessLevel;
+import lombok.Setter;
+import org.hibernate.Criteria;
+import org.openmrs.Obs;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.fhir2.FhirConstants;
+import org.openmrs.module.fhir2.api.dao.FhirMediaDao;
+import org.openmrs.module.fhir2.api.search.param.SearchParameterMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@Setter(AccessLevel.PACKAGE)
+public class FhirMediaDaoImpl extends BaseFhirDao<Obs> implements FhirMediaDao {
+
+//	private ObsService obsService = Context.getObsService();
+
+	@Autowired
+	private ObsService obsService;
+
+	@Override
+	public Obs get(@Nonnull String uuid) {
+		return obsService.getObsByUuid(uuid);
+	}
+	
+	@Override
+	public Obs createOrUpdate(@Nonnull Obs newEntry) {
+		return obsService.saveObs(newEntry, FhirConstants.SAVED_SUCCESSFULLY);
+	}
+	
+	@Override
+	public Obs delete(@Nonnull String uuid) {
+		return super.delete(uuid);
+	}
+	
+	@Override
+	public List<Obs> getSearchResults(@Nonnull SearchParameterMap theParams, @Nonnull List<String> resourceUuids) {
+		return resourceUuids.stream().map(obsService::getObsByUuid).collect(Collectors.toList());
+	}
+	
+	@Override
+	protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams) {
+		theParams.getParameters().forEach(entry -> {
```

Comment:
Here, we'll want to ensure that we're only searching ComplexObs and not just any Obs. This looks like it could be done as something like:

```java
if (lacksAlias(criteria, "c")) {
	criteria.createAlias("concept", "c");
}

if (lacksAlias(criteria, "cdt")) {
    criteria.createAlias("c.datatype", "cdt");
}

criteria.add(eq("cdt.hl7_abbreviation", "ED"));
```